src/node_url.cc

```diff
@@ -853,9 +868,20 @@ static url_host_type ParseHost(url_host* host,
   // First, we have to percent decode
   PercentDecode(input, length, &decoded);
 
-  // Then we have to punycode toASCII
-  if (!ToASCII(&decoded, &decoded))
-    goto end;
+  // Match browser behavior for ASCII only domains
+  // and do not run them through ToASCII algorithm.
+  if (IsAllASCII(decoded)) {
+    // Lowercase ASCII domains
+    for (size_t n = 0; n < decoded.size(); n++) {
+      if (!IsLowerCaseASCII(decoded[n])) {
```

Comment:
Won't this prevent idempotent writes to lowercase chars?. @joyeecheung  that was the intent  right?.